sys/kern/src/arch/arm_m.rs

```diff
@@ -309,15 +309,14 @@ pub fn reinitialize(task: &mut task::Task) {
         .iter()
         .find(|region| region.contains(initial_stack.saturating_sub(4)))
     {
-        let mut uslice: USlice<u32> = USlice::from_raw(
+        if let Ok(mut uslice) = USlice::<u32>::from_raw(
             region.base as usize,
             (initial_stack - frame_size - region.base as usize) >> 2,
-        )
-        .unwrap_lite();
-
-        let zap = task.try_write(&mut uslice).unwrap_lite();
-        for word in zap.iter_mut() {
-            *word = 0xbaddcafe;
+        ) {
+            let zap = task.try_write(&mut uslice).unwrap_lite();
```

Comment:
AFAICT, the `try_write` call failing here would be Bad News --- if a task's stack region isn't actual memory or that memory isn't writable, something has gone seriously wrong, so I think unwrapping this is reasonable. but, that might not be immediately obvious to a reader, who would see that we allow the `Uslice::from_raw` call to fail, but if we can't write to it, we throw up our hands and kill the whole system, which might look weird at a glance. Maybe a comment describing the difference between the two failure modes would be worth including here?

Reply:
Yeah, that was approximately my thinking with leaving the unwrap, but you're right that a comment would improve things.